### ISSUE TITLE
support quotas

### ISSUE THREAD

**Issue by travisghansen** [nmaupu/freenas-provisioner#3]
The more I've thought about this approach dataset + nfs share the more I like it.  I'd really like to start exploring supporting setting a quota on the dataset when created but am having troubles getting the build to work.  Being new to go makes this a bit of a stretch, but I'd like to help.  If you've got some pointers on building and running this from a development perspective that would be great.

**Comment by travisghansen** [comment]
```panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=1 addr=32 pc=94347298903359]

goroutine 1 [running]:
panic
	/build/gcc/src/gcc/libgo/go/runtime/panic.go:557
path.StripVendor..func1
	/build/glide/src/build/src/github.com/Masterminds/glide/path/strip.go:28
filepath.walk
	/build/gcc/src/gcc/libgo/go/path/filepath/path.go:377
filepath.walk
	/build/gcc/src/gcc/libgo/go/path/filepath/path.go:381
filepath.walk
	/build/gcc/src/gcc/libgo/go/path/filepath/path.go:381
filepath.walk
	/build/gcc/src/gcc/libgo/go/path/filepath/path.go:381
filepath.walk
	/build/gcc/src/gcc/libgo/go/path/filepath/path.go:381
path_filepath.Walk
	/build/gcc/src/gcc/libgo/go/path/filepath/path.go:403
github_com_Masterminds_glide_path.StripVendor
	/build/glide/src/build/src/github.com/Masterminds/glide/path/strip.go:22
github_com_Masterminds_glide_action.Install
	/build/glide/src/build/src/github.com/Masterminds/glide/action/install.go:62
main.commands..func12
	/build/glide/src/build/src/github.com/Masterminds/glide/glide.go:510
github_com_Masterminds_glide_vendor_github_com_codegangsta_cli.HandleAction
	/build/glide/src/build/src/github.com/Masterminds/glide/vendor/github.com/codegangsta/cli/app.go:490
github_com_Masterminds_glide_vendor_github_com_codegangsta_cli.Command.Run
	/build/glide/src/build/src/github.com/Masterminds/glide/vendor/github.com/codegangsta/cli/command.go:210
github_com_Masterminds_glide_vendor_github_com_codegangsta_cli.App.Run
	/build/glide/src/build/src/github.com/Masterminds/glide/vendor/github.com/codegangsta/cli/app.go:255
main.main
	/build/glide/src/build/src/github.com/Masterminds/glide/glide.go:104
make: *** [Makefile:27: vendor] Error 2
```

**Comment by nmaupu** [comment]
Nice !
Do you succeeded in compiling ? It seems that it's an error when running, isn't it ?
Also, the code currently support only in-cluster configuration so you will need a Kubernetes cluster to run it (minikube should do it).

Go dependencies are quite tricky specially trying to build stuff around Kubernetes SDK :/

What OS are you using ?

**Comment by nmaupu** [comment]
Ok, I built it successfully using `glide 0.13.0` and go `1.10` using:
```
make clean
make vendor
make
```
Don't use `glide up` command as previously stated, it's messing up the `vendor` directory.

**Comment by travisghansen** [comment]
@nmaupu yeah, that's what I've been doing.  On arch linux here.

```
go version
go version go1.10.3 linux/amd64
```
```
glide --version
glide version 0.13.0-dev
```

What's your setup.  I'm sure I can sandbox it in some way if necessary.

**Comment by nmaupu** [comment]
You could try some go building docker image.
Also, I know that go is very sensible about changing path where it has been configured for (check `GOPATH` and `GOROOT` env var).

**Comment by travisghansen** [comment]
@nmaupu yeah, I've been using a docker image for development just to build.  I'm sure I'll get to the bottom of it eventually on my host machine.

**Comment by travisghansen** [comment]
Closed by PR #6
